src/mono/wasm/runtime/loader/blazor/_Integration.ts

```diff
@@ -157,6 +157,43 @@ export function mapBootConfigToMonoConfig(moduleConfig: MonoConfigInternal, appl
         };
         assets.push(asset);
     }
+    for (let i = 0; i < resourceLoader.bootConfig.config.length; i++) {
+        let config = resourceLoader.bootConfig.config[i];
+        if (config === "appsettings.json" || config === `appsettings.${applicationEnvironment}.json`) {
+            const asset: AssetEntry = {
+                name: config,
+                resolvedUrl: `_framework/${name}`,
+                behavior: "vfs",
+            };
+
+            let response: Promise<Response> | null = null;
+            if (customLoadBootResource) {
+                const customLoadResult = customLoadBootResource("configuration", config, config, "");
+                if (customLoadResult instanceof Promise<Response>) {
+                    // They are supplying an entire custom response, so just use that
+                    response = customLoadResult;
+                } else if (typeof customLoadResult === "string") {
+                    // They are supplying a custom URL, so use that with the default fetch behavior
+                    config = customLoadResult;
+                }
+            }
+
+            if (!response) {
+                response = fetch(config, {
```

Comment:
This `fetch` would start too early. I prefer that `assets.ts` drives the timing.

Reply:
Also, why not just `resolvedUrl` and delegate the responsibility ?
- fetch_like
- cache
...